cattle/plugins/docker/__init__.py

```diff
@@ -60,7 +60,7 @@ def is_host_pidns():
         return default_value('AGENT_PIDNS', 'container') == 'host'
 
 
-def docker_client(version=None, base_url_override=None):
+def docker_client(version=None, base_url_override=None, tls_config=None):
```

Comment:
I'm commenting on old code here, but are these first two ifs really mutually exclusive?
```
     if DockerConfig.use_boot2docker_connection_env_vars():
         kwargs = kwargs_from_env(assert_hostname=False)
     elif base_url_override is not None:
         kwargs = {'base_url': base_url_override}
```
The use_boot... if statement is to allow python-agent to work from your laptop. It is mainly for when you want to run the py-agent tests locally. So, when running locally, if base_url_override is passed in, it ***should*** still override the b2d connection params because you actually want to connect to a different remote host.

Also, keep in mind that kwargs_from_env sets the tls config. So, you could get into a weird scenario where it is set there and then overridden by the tls_config passed in.

Finally, I think both `elif base_url_override is not None:` or `if tls_config is not None:` should drop the `is not None` to catch all non-zero-value scenarios (empty dicts and blank strings specifically).